crates/plugins/mediator-coordination/src/plugin.rs

```diff
@@ -55,28 +58,21 @@ impl Plugin for MediatorCoordinationPlugin {
         "mediator_coordination"
     }
 
-    // Load environment variables required for this plugin
-
     fn mount(&mut self) -> Result<(), PluginError> {
-        let MediatorCoordinationPluginEnv {
-            storage_dirpath,
-            mongo_uri,
-            mongo_dbn,
-            ..
-        } = load_plugin_env()?;
+        let env = load_plugin_env()?;
 
         // Expect DID document from file system
-        if did_endpoint::validate_diddoc(&storage_dirpath).is_err() {
+        if did_endpoint::validate_diddoc(&env.storage_dirpath).is_err() {
             tracing::error!("diddoc validation failed; is plugin did-endpoint mounted?");
             return Err(PluginError::InitError);
         }
 
         // Check connectivity to database
+        let db = load_mongo_connector(&env.mongo_uri, &env.mongo_dbn)?;
 
-        if load_mongo_connector(&mongo_uri, &mongo_dbn).is_err() {
-            tracing::error!("could not establish connectivity with mongodb");
-            return Err(PluginError::InitError);
-        }
+        // Save the environment and MongoDB connection in the struct
```

Comment:
No you can save actuall values to be reused in a different function in a self parameter

Reply:
Yes, You can. In my case,
When i store the environment (env) and database connection (db) in the self struct in mount, i can access them later in other functions by referencing self.

Reply:
Yes, You can.In my case, 
When i store the environment (env) and database connection (db) in the self struct in mount, i can access them later in other functions by referencing self.

Reply:
No you are missing the point try to answer this questions
- what is self
- how do i create or have a self

Reply:
Hi @Christiantyemele. Did you notice these are later saved in `self`? Or am I coming late?
https://github.com/adorsys/didcomm-mediator-rs/blob/de535f1277d30b12f4b94db7a1f58daf37d971ee/crates/plugins/mediator-coordination/src/plugin.rs#L73-L75